Hdcpd 15

ljia5 opened the pull request.

ljia5 committed:
Enable media hdcp

1. Enable daemon for hdcp1.4/hdcp2.2
2. Add hdcp Test Utility

Tracked-On: OAM-130842
Signed-off-by: Jia, Lin A <lin.a.jia@intel.com>

ljia5 committed:
Change hdcpd build method

Change to Android.bp for hdcp build

Tracked-On: OAM-130842
Signed-off-by: Jia, Lin A <lin.a.jia@intel.com>

ljia5 committed:
Add virtio_gpu support for hdcpd

Open virtio_gpu drm device for acrn support

Tracked-On: OAM-130842
Signed-off-by: Jia, Lin A <lin.a.jia@intel.com>

sysopenci commented:
Program name for this pr is not compatable with other dependent prs, for more details please check tracked_on

sysopenci commented:
Program name for this pr is not compatable with other dependent prs, for more details please check tracked_on

sysopenci commented:
Program name for this pr is not compatable with other dependent prs, for more details please check tracked_on

sysopenci commented:
Program name for this pr is not compatable with other dependent prs, for more details please check tracked_on

ljia5 committed:
Fix hdcpd build fail for android15

1. Remove hdcpd srm function due to avoid hdcpd build fail
2. Delete unused build code

Tracked-On: OAM-130842
Signed-off-by: Jia, Lin A <lin.a.jia@intel.com>

sysopenci commented:
Program name for this pr is not compatable with other dependent prs, for more details please check tracked_on

feijiang1 commented on daemon/Android.mk:
```diff
@@ -25,6 +25,7 @@ include $(CLEAR_VARS)
 
 LOCAL_CPPFLAGS := \
     -DANDROID \
+    -DUSES_IA_HWCOMPOSER \
```
why still need use USES_IA_HWCOMPOSER?

feijiang1 commented on daemon/portmanager_android.cpp:
```diff
@@ -156,13 +159,15 @@ int32_t setPortProperty_hwcservice(int32_t m_DrmFd,
            {
-	       HDCP_ASSERTMESSAGE("Attempting HDCP Enable");
+	       HDCP_ASSERTMESSAGE("Attempting HDCP Enable, value =%d", value);
+	       HDCP_ASSERTMESSAGE("Attempting HDCP Enable, propValue =%d", propValue);
```
can combine together as one message

feijiang1 commented on daemon/portmanager_android.cpp:
```diff
@@ -156,13 +159,15 @@ int32_t setPortProperty_hwcservice(int32_t m_DrmFd,
                //This is only for HDCP2.2
-	       HDCP_ASSERTMESSAGE("Attempting HDCP Enable Type 1");
+	       HDCP_ASSERTMESSAGE("Attempting HDCP Enable Type, value =%d", value);
+	       HDCP_ASSERTMESSAGE("Attempting HDCP Enable Type, propValue =%d", propValue);
```
can combine together as one message

feijiang1 commented on test/CMakeLists.txt:
```diff
@@ -0,0 +1,22 @@
+# OTHER DEALINGS IN THE SOFTWARE.
+
+
+add_subdirectory(hdcpTestUtility)
```
why need this?

feijiang1 commented on test/hdcpTestUtility/CMakeLists.txt:
```diff
@@ -0,0 +1,30 @@
+
+target_compile_options(${PROJECT_NAME} PRIVATE ${HDCP_CXX_FLAGS})
+
+target_link_libraries(${PROJECT_NAME} hdcpsdk)
```
it it necessary?

feijiang1 commented on test/hdcpTestUtility/hdcpTestUtility.cpp:
```diff
@@ -0,0 +1,339 @@
+{
+    if (nullptr == ctx)
+    {
+        printf("hdcp context pointer is null\n");
```
should use LOGV?

ljia5 committed:
Refine hdcpd and hdcpTestutily

1. Remove unsed source code
2. Rename USES_IA_HWCOMPOSER define

Tracked-On: OAM-130842
Signed-off-by: Jia, Lin A <lin.a.jia@intel.com>

sysopenci commented:
Android CI has started Engineering Build for this issue ,Please check the linked Tracked-On issue/Android CI Web  for more details.

feijiang1 reviewed (approved):
LGTM

sysopenci commented:
Android CI has started Engineering Build for this issue ,Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
FAILURE: Android CI has completed Engineering Build for this issue.Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
Android CI has started MERGE Build for this pr ,Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
Android CI has completed MERGE Build for this pr, build is FAILURE. Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
Android CI has started MERGE Build for this pr ,Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
Android CI has completed MERGE Build for this pr, build is FAILURE. Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
Android CI has started MERGE Build for this pr ,Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
Android CI has completed MERGE Build for this pr, build is SUCCESS. Please check the linked Tracked-On issue/Android CI Web  for more details. For Binaries: /cactus-absp-or-local/celadon-merge/1087
